### ISSUE TITLE
Validation set handling

### ISSUE THREAD

**Issue by adabbott** [CCQC/PES-Learn#14]
* Support validation sets both from externally supplied datasets and PES-Learn datasets for GPs
* Clean up neural network validation set specifications
* Look into adding cross validation
